add(epp): outscale (#2230)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/cloud-outscale.md

---
id: cloud-outscale
title: Outscale API
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack Assets

### Templates

> The Centreon Pack **Outscale** brings a host template:

Reviewer: ```suggestion
The Centreon Monitoring Connector **Outscale** brings a host template:
```

| volume attachment status |       |

</TabItem>
<TabItem value="Vpn-Connections" label="Vpn-Connections">

| Metric Name                     | Unit  |
|:--------------------------------|:------|
| vpn_connections.detected.count  |       |
| vpn_connections.available.count |       |
| vpn_connections.pending.count   |       |
| vpn_connections.deleting.count  |       |
| vpn_connections.deleted.count   |       |
| vpn connection status           |       |

</TabItem>
</Tabs>

## Prerequisites

Please follow the official documentation to install ```osc-cli``` for the user ```centreon-engine```:
https://docs.outscale.com/en/userguide/Installing-and-Configuring-OSC-CLI.html

A user Outscale with the following privileges should be used:
```
    "Statement": [
        {
            "Action": [
                "*:Describe*",   ç Les droits sont bien positionnés…
                "*:Read*"
            ],
            "Resource": "*",
            "Effect": "Allow"
        }
    ]
}
```

## Setup

### Monitoring Pack

If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the pack displayed within the

> **Configuration > Plugin Packs > Manager** menu.

Reviewer: ```suggestion
**Configuration > Monitoring Connectors Manager** menu.
```

If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

> 

Reviewer: EL9?

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-cloud-outscale
```

> </TabItem>
> <TabItem value="CentOS 7" label="CentOS 7">

Reviewer 2: ```suggestion
</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-cloud-outscale
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">
```

```bash
yum install centreon-pack-cloud-outscale
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-pack-cloud-outscale
```

</TabItem>
</Tabs>

Whatever the license type (*online* or *offline*), install the **Outscale** Pack through

> the **Configuration > Plugin Packs > Manager** menu.

Reviewer: ```suggestion
the **Configuration > Monitoring Connectors Manager** menu.
```

https://docs.outscale.com/en/userguide/Installing-and-Configuring-OSC-CLI.html

A user Outscale with the following privileges should be used:
```
    "Statement": [
        {
            "Action": [
                "*:Describe*",   ç Les droits sont bien positionnés…
                "*:Read*"
            ],
            "Resource": "*",
            "Effect": "Allow"
        }
    ]
}
```

## Setup

### Monitoring Pack

If the platform uses an *online* license, you can skip the package installation
instruction below as it is not required to have the pack displayed within the
**Configuration > Monitoring Connectors Manager** menu.
If the platform uses an *offline* license, install the package on the **central server**
with the command corresponding to the operating system's package manager:

<Tabs groupId="sync">
<TabItem value="Alma / RHEL / Oracle Linux 8" label="Alma / RHEL / Oracle Linux 8">

```bash
dnf install centreon-pack-cloud-outscale
```

</TabItem>
<TabItem value="Alma / RHEL / Oracle Linux 9" label="Alma / RHEL / Oracle Linux 9">

```bash
dnf install centreon-pack-cloud-outscale

> ```bash

Reviewer: Centos7 tab has been removed.

yum install centreon-pack-cloud-outscale
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-pack-cloud-outscale
```

</TabItem>
</Tabs>

Whatever the license type (*online* or *offline*), install the **Outscale** Pack through
the **Configuration > Monitoring Connectors Manager** menu.

### Plugin

Since Centreon 22.04, you can benefit from the 'Automatic plugin installation' feature.
When this feature is enabled, you can skip the installation part below.

You still have to manually install the plugin on the poller(s) when:
- Automatic plugin installation is turned off
- You want to run a discovery job from a poller that doesn't monitor any resource of this kind yet

> More information in the [Installing the plugin](/docs/monitoring/pluginpacks/#installing-the-plugin) section.

Use the commands below according to your operating system's package manager:

> 

Reviewer: EL9 tab still missing.
